x-pack/plugins/security_solution/common/entity_analytics/risk_engine/privileges.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { EntityAnalyticsPrivileges } from '../../api/entity_analytics/common';
+import {
+  RISK_ENGINE_REQUIRED_ES_CLUSTER_PRIVILEGES,
+  RISK_ENGINE_REQUIRED_ES_INDEX_PRIVILEGES,
+} from './constants';
+
+export type MissingClusterPrivileges = string[];
+export type MissingIndexPrivileges = Array<[indexName: string, privileges: string[]]>;
```

Comment:
Nit: I _believe_ it's a general Typescript best practice to make tuples readonly when possible. The array itself won't be readonly, but the tuples contained within it would be so that someone can't accidentally `tuple.push(something)`).

```suggestion
export type MissingIndexPrivileges = Array<readonly [indexName: string, privileges: string[]]>;
```

Reply:
Also feel free to ignore this comment if it's not applicable 😄